cpp/lib/src/ProcessUtil.cc

```diff
@@ -0,0 +1,49 @@
+/** \file    ProcessUtil.c
+ *  \brief   The ProcessUtil implementation.
+ *  \author  Dr. Johannes Ruscheinski
+ */
+
+/*
+ *  Copyright 2020 University Library of Tübingen
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "ProcessUtil.h"
+#include "FileUtil.h"
+#include "StringUtil.h"
+
+
+namespace ProcessUtil {
+
+
+std::unordered_set<pid_t> GetProcessIdsForPath(const std::string &path) {
+    std::unordered_set<pid_t> pids;
+
+    FileUtil::Directory proc("/proc", "\\d+"); // Only PID's
+    for (const auto &pid_dir : proc) {
+        FileUtil::Directory fd_dir("/proc/" + pid_dir.getName() + "/fd");
```

Comment:
due to pid_dir.getName() documentation, it returns the full name of the entry including the directory path, so prefixing it with "/proc" should not be necessary.

Reply:
That documentation is incorrect! :-( (I fixed it.)